pkg/api/v1alpha2/types_config.go

```diff
@@ -0,0 +1,169 @@
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ImageSetConfiguration object kind.
+const ImageSetConfigurationKind = "ImageSetConfiguration"
+
+// ImageSetConfiguration configures image set creation.
+type ImageSetConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+	// ImageSetConfigurationSpec defines the global configuration for an imageset.
+	ImageSetConfigurationSpec `json:",inline"`
+}
+
+// ImageSetConfigurationSpec defines the global configuration for an imageset.
+type ImageSetConfigurationSpec struct {
+	// Mirror defines the configuration for content types within the imageset.
+	Mirror Mirror `json:"mirror"`
+	// ArchiveSize is the size of the segmented archive in GB
+	ArchiveSize int64 `json:"archiveSize,omitempty"`
+	// StorageConfig for reading/writing metadata and files.
+	StorageConfig StorageConfig `json:"storageConfig"`
+}
+
+// Mirror defines the configuration for content types within the imageset.
+type Mirror struct {
+	// OCP defines the configuration for OCP and OKD content types.
+	OCP OCP `json:"ocp,omitempty"`
+	// Operators defines the configuration for Operator content types.
+	Operators []Operator `json:"operators,omitempty"`
+	// AdditionalImages defines the configuration for a list
+	// of individual image content types.
+	AdditionalImages []AdditionalImages `json:"additionalImages,omitempty"`
+	// Helm define the configuration for Helm content types.
+	Helm Helm `json:"helm,omitempty"`
+	// BlockedImages define a list of images that will be blocked
+	// from the mirroring process if they exist in other content
+	// types in the configuration.
+	BlockedImages []BlockedImages `json:"blockedImages,omitempty"`
+	// Samples defines the configuration for Sample content types.
+	// This is currently not implemented.
+	Samples []SampleImages `json:"samples,omitempty"`
+}
+
+// OCP defines the configuration for OCP and OKD content types.
+type OCP struct {
```

Comment:
Nit: have you considered renaming this to something more readable, and less OCP related, for example `Product`? Also do you care about the distinction if that's OCP or OKD, if so maybe consider adding enum type allowing to discriminate that.

Reply:
I agree that the name of the type could be more general because we are supporting both OCP and OKD. We make the distinction between OCP and OKD from the channel name the OKD channel name is distinct.